include/wabt/type-checker.h

```diff
@@ -44,6 +65,75 @@ class TypeChecker {
     Index type_index;
   };
 
+  struct StructType {
+    StructType() = default;
+    StructType(const TypeMutVector& fields) : fields(fields) {}
+
+    TypeMutVector fields;
+  };
+
+  struct ArrayType {
+    ArrayType() = default;
+    ArrayType(TypeMut field) : field(field) {}
+
+    TypeMut field;
+  };
+
+  struct RecursiveRange {
```

Comment:
Maybe we should use `RecGroup` and `rec_group` rather then `Recursive` or `rec` ?   Or maybe I'm just not used to seeing "rec" / "recurisive" and thinking of recursion group?